apps/devhub/utils.py

```diff
@@ -67,8 +67,14 @@ def limit_validation_results(validation):
     if lim and len(messages) > lim:
         # Sort messages by severity first so that the most important messages
         # are the one we keep.
-        TYPES = {'error': 0, 'warning': 1, 'notice': 2}
-        messages.sort(key=lambda msg: TYPES.get(msg.get('type')))
+        TYPES = {'error': 0, 'warning': 2, 'notice': 3}
+
+        def message_key(message):
+            if message.get('signing_severity'):
```

Comment:
Are you testing for the presence of the `signing_severity` key here? If so, using `'signing_severity' in message` might be clearer?